charts/orchestrator/templates/tekton-pipeline.yaml

```diff
@@ -0,0 +1,164 @@
+      type: string
+      default: "true"
+  workspaces:
+    - name: workflow-source
+    - name: workflow-config
+    - name: ssh-creds
+    - name: docker-credentials
+  tasks:
+    - name: fetch-workflow
+      taskRef:
+        name: git-cli
+      workspaces:
+        - name: source
+          workspace: workflow-source
+        - name: ssh-directory
+          workspace: ssh-creds
+      params:
+        - name: GIT_USER_NAME
+          value: Parodos Tekton Pipeline
+        - name: GIT_USER_EMAIL
+          value: parodos@redhat.com
+        - name: USER_HOME
+          value: /home/git
+        - name: GIT_SCRIPT
+          value: |
+            eval "$(ssh-agent -s)"
+            ssh-add "${PARAM_USER_HOME}"/.ssh/id_rsa
+            git clone $(params.gitUrl) workflow
+            cd workflow
+    - name: fetch-workflow-config
+      taskRef:
+        name: git-cli
+      workspaces:
+        - name: source
+          workspace: workflow-config
+        - name: ssh-directory
+          workspace: ssh-creds
+      params:
+        - name: GIT_USER_NAME
+          value: Parodos Tekton Pipeline
+        - name: GIT_USER_EMAIL
+          value: parodos@redhat.com
+        - name: USER_HOME
+          value: /home/git
+        - name: GIT_SCRIPT
+          value: |
+            eval "$(ssh-agent -s)"
+            ssh-add "${PARAM_USER_HOME}"/.ssh/id_rsa
+            git clone $(params.gitConfigUrl) workflow-config
+    - name: flatten-workflow
+      runAfter: ["fetch-workflow"]
+      taskRef:
+        name: flattener
+      workspaces:
+        - name: workflow-source
+          workspace: workflow-source
+      params:
+        - name: workflowId
+          value: $(params.workflowId)
+        - name: convertToFlat
+          value: $(params.convertToFlat)
+    - name: build-manifests
+      runAfter: ["flatten-workflow"]
+      taskRef:
+        name: build-manifests
+      workspaces:
+        - name: workflow-source
+          workspace: workflow-source
+      params:
+        - name: workflowId
+          value: $(params.workflowId)
+    - name: build-config
+      runAfter: ["build-manifests", "fetch-workflow-config"]
+      taskRef:
+        name: build-config
+      workspaces:
+        - name: workflow-source
+          workspace: workflow-source
+        - name: workflow-config
+          workspace: workflow-config
+      params:
+        - name: workflowId
+          value: $(params.workflowId)
+        - name: imageTag
+          value: $(tasks.fetch-workflow.results.commit)
+    - name: build-and-push-image
+      runAfter: ["flatten-workflow"]
+      taskRef:
+        name: buildah
+        kind: ClusterTask        
+      workspaces:
+        - name: source
+          workspace: workflow-source
+        - name: dockerconfig
+          workspace: docker-credentials
+      params:
+        - name: IMAGE
+          value: quay.io/orchestrator/serverless-workflow-$(params.workflowId):$(tasks.fetch-workflow.results.commit)
```

Comment:
need to document this somewhere - to enable users to use their image registries.

Reply:
https://github.com/parodos-dev/orchestrator-helm-chart/issues/96